[OSDOCS-5344]: Adding concepts and personas for hosted control planes (#61057)
openshift/openshift-docs: modules/hosted-control-planes-concepts-personas.adoc

// Module included in the following assemblies:
//
// * architecture/control-plane.adoc
// * hosted-control-planes/index.adoc

:_content-type: CONCEPT
[id="hosted-control-planes-concepts-personas_{context}"]
= Glossary of common concepts and personas for hosted control planes

When you use hosted control planes for {product-title}, it is important to understand its key concepts and the personas that are involved.

> Management cluster:: An {product-title} cluster that hosts hosted control planes and zero to many hosted clusters.

Reviewer: ```suggestion
Management cluster:: An {product-title} cluster where: 

- The HyperShift operator is deployed. 
- Where the control planes for tenant clusters are hosted. 

```

> Hosted cluster:: An {product-title} API endpoint that is managed by hosted control planes.

Reviewer: ```suggestion
Hosted cluster:: An {product-title} API endpoint for the tenant cluster managed by the HyperShift operator. 
```

Hosted control plane:: An {product-title} control plane that runs on the management cluster, which is exposed by the API endpoint of a hosted cluster. The components of a control plane include etcd, an API server, the Kubernetes controller manager, and a VPN.

Hosted control plane infrastructure:: Resources on the management cluster or external cloud provider that are prerequisites to run hosted control plane processes.

Management cluster infrastructure:: Network, compute, and storage resources of the management cluster.

> Hosted cluster infrastructure:: Network, compute, and storage resources that exist in the customer cloud account.

Reviewer: ```suggestion
Hosted cluster infrastructure:: Network, compute, and storage resources that exist in the tenant/end-user cloud account.
```

> Cluster service provider:: The user who hosts the cluster control planes and is responsible for uptime, observing the user interface for fleet-wide alerts. The cluster service provider configures the cloud account to host control planes, the user-provisioned infrastructure, which includes the host awareness of available compute, and where VMs are pulled from. This user has cluster administrator management access.

Reviewer: ```suggestion
Cluster service provider:: The role assumed by the user who hosts the cluster control planes and is responsible for uptime, observing the user interface for fleet-wide alerts. The cluster service provider persona configures the cloud account for the management cluster to host control planes, the user-provisioned infrastructure, which includes the host awareness of available compute, and where VMs are pulled from. Assuming this role provides cluster administrator access to the management cluster.
```

Reviewer: I still think the sentence is too long.

Reviewer: Adding the following will also help highlight and explain the scope: 

Users assuming this role are usually cluster-admins on the management cluster and have RBAC to monitor and own the availability of the HyperShift operator as well as the control planes for the tenant’s hosted clusters.

> Cluster service consumer:: Th user who can request control planes and workers, drive updates, or modify externalized configurations. Typically, this user does not manage or access cloud credentials or infrastructure encryption keys.

Reviewer: ```suggestion
Cluster service consumer:: The role assumed by users who can request control planes and workers, drive updates, or modify externalized configurations. Typically, this user does not manage or access cloud credentials or infrastructure encryption keys.
```

Reviewer: Adding the following will also help highlight and explain the scope for the `Cluster Service Consumer`:

Can request Hosted Clusters and interact with NodePools (to upgrade,...). Users assuming this role have RBAC to CRUD HC/NodePools within some logical boundary.

> Cluster instance administrator:: The user who has the `cluster-admin` role in the provisioned cluster, but who might not have power over when or how the cluster is updated or configured. This user might have read-only access to see some configuration projected into the cluster.

Reviewer: ```suggestion
Cluster instance administrator:: The role assumed by the user who has the `cluster-admin` role in the provisioned cluster, but who might not have power over when or how the cluster is updated or configured. This user might have read-only access to see some configuration projected into the cluster.
```

Reviewer: We might add "who is the equivalent to the administrator in stand-alone OpebShift" to be consistent with the instance developer.

> Cluster (instance) user:: A user who is the equivalent of a developer in stand-alone {product-title}. This user does not have a view into OperatorHub or machines.

Reviewer: ```suggestion
Cluster (instance) user:: The role assumed by the user who is the equivalent of a developer in stand-alone {product-title}. This user does not have a view into OperatorHub or machines.
```

[id="hosted-control-planes-concepts-personas_{context}"]
= Glossary of common concepts and personas for hosted control planes

When you use hosted control planes for {product-title}, it is important to understand its key concepts and the personas that are involved.

Management cluster:: An {product-title} cluster where the HyperShift Operator is deployed and where the control planes for tenant clusters are hosted.

Hosted cluster:: An {product-title} API endpoint for the tenant cluster that is managed by the HyperShift Operator.

Hosted control plane:: An {product-title} control plane that runs on the management cluster, which is exposed by the API endpoint of a hosted cluster. The components of a control plane include etcd, an API server, the Kubernetes controller manager, and a VPN.

Hosted control plane infrastructure:: Resources on the management cluster or external cloud provider that are prerequisites to run hosted control plane processes.

Management cluster infrastructure:: Network, compute, and storage resources of the management cluster.

Hosted cluster infrastructure:: Network, compute, and storage resources that exist in the tenant or end-user cloud account.

> Cluster service provider:: This role is assumed by a user who hosts the cluster control planes. Typically, users who assume this role are cluster administrators on the management cluster and have Role-Based Access Control to monitor and own the availability of the HyperShift Operator and the control planes for the tenant's hosted clusters. The cluster service provider persona is responsible for the following activities:

Reviewer: Should have probably been my first suggestion (not in a comment) earlier. 

```suggestion
Cluster service provider:: Users assuming this role are usually cluster-admins on the management cluster and have RBAC to monitor and own the availability of the HyperShift operator as well as the control planes for the tenant’s hosted clusters. 
 The cluster service provider persona is responsible for the following activities:
```

> ** Maintaining uptime by observing the user interface for fleet-wide alerts

Reviewer: This is not the whole set but a subset (i.e., including but not limited to).

> ** Maintaining uptime by observing the user interface for fleet-wide alerts

Reviewer: ```suggestion
** Maintaining uptime by observing the user interface for fleet-wide alerts
```

Which user interface is not that clear.

 We could say, owning service-level objects for control-planes availability, uptime, stability, ...

** Configuring the cloud account for the management cluster to host control planes
** Configuring the user-provisioned infrastructure, which includes the host awareness of available compute
** Configuring where VMs are pulled from

> Cluster service consumer:: The role that is assumed by users who can request control planes and workers, drive updates, or modify externalized configurations. Typically, this user does not manage or access cloud credentials or infrastructure encryption keys. The cluster service consumer persona can request hosted clusters and interact with node pools. Users who assume this role have Role-Based Access Control to create, read, update, or delete hosted clusters and node pools within a logical boundary.

Reviewer: ```suggestion
Cluster service consumer:: Users assuming this role can request control planes and workers, drive updates, or modify externalized configurations. Typically, this user does not manage or access cloud credentials or infrastructure encryption keys. The cluster service consumer persona can request hosted clusters and interact with node pools. Users who assume this role have Role-Based Access Control to create, read, update, or delete hosted clusters and node pools within a logical boundary.
```

// Module included in the following assemblies:
//
// * architecture/control-plane.adoc
// * hosted-control-planes/index.adoc

:_content-type: CONCEPT
[id="hosted-control-planes-concepts-personas_{context}"]
= Glossary of common concepts and personas for hosted control planes

When you use hosted control planes for {product-title}, it is important to understand its key concepts and the personas that are involved.

[id="hosted-control-planes-concepts_{context}"]
== Concepts

hosted cluster:: An {product-title} API endpoint for the tenant cluster that is managed by the HyperShift Operator.

hosted cluster infrastructure:: Network, compute, and storage resources that exist in the tenant or end-user cloud account.

hosting cluster:: See _management cluster_.

> hosted control plane:: An {product-title} control plane that runs on the management cluster, which is exposed by the API endpoint of a hosted cluster. The components of a control plane include etcd, an API server, the Kubernetes controller manager, and a VPN.

Author: @zanetworker Do you know the answer to Andrea's question?

> hosted control plane:: An {product-title} control plane that runs on the management cluster, which is exposed by the API endpoint of a hosted cluster. The components of a control plane include etcd, an API server, the Kubernetes controller manager, and a VPN.

Reviewer: Sure, better always be explicity. That's the kubernetes API server.

> Hosted control plane infrastructure:: Resources on the management cluster or external cloud provider that are prerequisites to run hosted control plane processes.

Reviewer: While this can provide some entropy, I feel we can ommit to avoid confusion. We already talk about infrastructure with the management cluster, so can be considered redundant.

Reviewer: Not the same thing. The hosting cluster is the management cluster. I.e., the cluster that does the hosting act for the control planes. The hosted cluster is the acted upon, that's the cluster provisioned by the cluster service. Hope that makes sense :)

Reviewer: So we can probably add a synonym to the management cluster because management/hosting tend to be used interchangeably.

> management cluster:: An {product-title} cluster where the HyperShift Operator is deployed and where the control planes for tenant clusters are hosted. In some contexts, the management cluster is referred to as the _hosting cluster_.

Reviewer 2: Not critical, but if these aren't totally interchangeable, just wondering if it would be worthwhile pointing out which contexts it's referred to as "hosting cluster" (and specify that in the "hosting cluster" definition, instead of just pointing here.

Author: They are interchangeable. I'll update the management cluster entry to make that clearer.

management cluster infrastructure:: Network, compute, and storage resources of the management cluster.

[id="hosted-control-planes-personas_{context}"]

> == Personas

Reviewer 2: One question I have about these personas are if they are just personas/typical ways certain users use hosted control planes) or if they are actually official roles that are set somewhere. The way is reads now to me is the former, but then it talks about legitimate RBAC permissions, etc. that makes me think they might be real roles (and if they are, then it should probably list its actual official role, like how we do `cluster-admin`, etc.).

Reviewer: Not real roles. The real roles are the things that the personas can assume, for example, cluster-admin. This is more "thought leadership" and alignment to allow us to use common language to discuss use-case. Also what we are pushing upstream.

> cluster instance administrator:: The role that is assumed by the user who is the equivalent to the administrator in stand-alone {product-title}. This user has the cluster-admin role in the provisioned cluster, but might not have power over when or how the cluster is updated or configured. This user might have read-only access to see some configuration projected into the cluster.

Reviewer 2: * "standalone" has no hyphen (per SSG's [standalone entry](https://redhat-documentation.github.io/supplementary-style-guide/#standalone)) (here and in the next definition too)
* cluster-admin in backticks

```suggestion
cluster instance administrator:: The role that is assumed by the user who is the equivalent to the administrator in standalone {product-title}. This user has the `cluster-admin` role in the provisioned cluster, but might not have power over when or how the cluster is updated or configured. This user might have read-only access to see some configuration projected into the cluster.
```

cluster instance user:: The role that is assumed by the user who is the equivalent of a developer in stand-alone {product-title}. This user does not have a view into OperatorHub or machines.

> cluster service consumer:: Users who assume this role can request control planes and workers, drive updates, or modify externalized configurations. Typically, this user does not manage or access cloud credentials or infrastructure encryption keys. The cluster service consumer persona can request hosted clusters and interact with node pools. Users who assume this role have RBAC to create, read, update, or delete hosted clusters and node pools within a logical boundary.

Reviewer 2: I don't think we normally say just "workers". It would usually be "control plane nodes and worker nodes", if that's what you mean here

cluster service provider:: Users who assume this role are usually cluster-admins on the management cluster and have RBAC to monitor and own the availability of the HyperShift Operator as well as the control planes for the tenant's hosted clusters. The cluster service provider persona is responsible for the several activities, including, but not limited to, the following examples:
** Owning service-level objects for control plane availability, uptime, and stability.
** Configuring the cloud account for the management cluster to host control planes

> ** Configuring the user-provisioned infrastructure, which includes the host awareness of available compute

Author: Good catch! It should be "compute resources".

> Cluster (instance) user:: The role that is assumed by the user who is the equivalent of a developer in stand-alone {product-title}. This user does not have a view into OperatorHub or machines.

Reviewer: Do we use (instance) or just "instance" consistently? I.e., Cluster instance administrator vs. Cluster (instance) administrator.

Author: Good catch! I copied that verbatim from the upstream docs, but you're right that we should be consistent. I did a few searches in the OpenShift docs as well as on Google, and I didn't find any hits for the specific phrases "cluster instance user" or "cluster instance administrator". That makes me think that we can probably omit the "instance" and just say "Cluster administrator" and "Cluster user". Do you know of any reason why we'd need to keep the "instance"?

Reviewer: Agree :)

Author: Thanks for the clarification! In that case, I'll leave "instance" in, without the parentheses for consistency.
